Lib/typing.py

```diff
@@ -3296,3 +3297,23 @@ def method(self) -> None:
         # read-only property, TypeError if it's a builtin class.
         pass
     return method
+
+
+def get_protocol_members(tp: type, /) -> set[str]:
+    """Return the set of members defined in a Protocol.
+
+    Example::
+
+        >>> from typing import Protocol, get_protocol_members
+        >>> class P(Protocol):
+        ...     def a(self) -> str: ...
+        ...     b: int
+        >>> get_protocol_members(P)
+        {'a', 'b'}
+
+    Return None for arguments that are not Protocols.
+
+    """
+    if not getattr(tp, '_is_protocol', False) or tp is Protocol:
+        return None
+    return tp.__protocol_attrs__
```

Comment:
We should return a copy, rather than the original set, or users might (accidentally or deliberately) mutate it:

```suggestion
    return frozenset(tp.__protocol_attrs__)
```

It would also be good to add a test to make sure that users can't mutate the attribute on the original class by using this function.